samples/reasoning_engine/create_embeddings.py

```
@@ -0,0 +1,103 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import asyncio
+import os
+import uuid
+
+from google.cloud import resourcemanager_v3  # type: ignore
+from langchain_community.document_loaders.csv_loader import CSVLoader
+from langchain_google_vertexai import VertexAIEmbeddings
+
+from langchain_google_cloud_sql_pg import PostgresEngine, PostgresVectorStore
+
+PROJECT_ID = os.getenv("PROJECT_ID") or "my-project-id"
+REGION = os.getenv("REGION") or "us-central1"
+INSTANCE = os.getenv("INSTANCE") or "my-primary"
+DATABASE = os.getenv("DATABASE") or "my_database"
+TABLE_NAME = os.getenv("TABLE_NAME") or "my_test_table"
+CHAT_TABLE_NAME = os.getenv("CHAT_TABLE_NAME") or "my_chat_table"
+USER = os.getenv("DB_USER") or "postgres"
+PASSWORD = os.getenv("DB_PASSWORD") or "password"
+
+
+async def create_databases():
+    engine = await PostgresEngine.afrom_instance(
+        PROJECT_ID,
+        REGION,
+        INSTANCE,
+        database="postgres",
+        user="postgres",
+        password=PASSWORD,
+    )
+    try:
+        await engine._aexecute_outside_tx(f"CREATE DATABASE {DATABASE}")
+    except Exception as e:
+        print(e)
```

Not sure we should catch this exception ?

---

I was going to create and delete the database each time but it's easier to delete tables from the database; therefore I don't really care about this error. The duplicate database error is hard to catch because the sqlalchemy changes it to just an OperationalError.

---

Ok -- slightly concerned that it'll mask other things like connectivity errors. I think at a minimum we should leave a comment why it's okay to hide this error.

This is how we handled in in the DB retrieval app fwiw: https://github.com/GoogleCloudPlatform/genai-databases-retrieval-app/blob/00d8ddd5d0c3df45c206ed89af4fb6a69bb7c19c/retrieval_service/datastore/providers/cloudsql_postgres_test.py#L81

We probably need to dispose the engine regardless?